samples/nrf9160/serial_lte_modem/src/gps/th_gps_service.c

```diff
@@ -0,0 +1,294 @@
+	LOG_DBG("This may take several minutes.");
+	LOG_DBG("The device will attempt to get a fix for %d seconds, ",
+		CONFIG_GPS_CONTROL_FIX_TRY_TIME);
+	LOG_DBG("before the GPS is stopped.");
+
+	return 0;
+}
+
+static int stop(void)
+{
+	int err;
+
+	err = gps_stop(gps_work.dev);
+	if (err) {
+		return err;
+	}
+
+	LOG_INF("GPS stopped successfully");
+	return 0;
+}
+
+static void gps_work_handler(struct k_work *work)
+{
+	int err;
+
+	if (gps_work.type == GPS_WORK_START) {
+		err = start();
+		if (err) {
+			LOG_ERR("GPS could not be started, error: %d", err);
+			return;
+		}
+
+		LOG_INF("GPS operation started");
+
+		atomic_set(&gps_is_active, 1);
+
+		gps_work.type = GPS_WORK_STOP;
+
+		k_delayed_work_submit(&gps_work.work,
+				K_SECONDS(CONFIG_GPS_CONTROL_FIX_TRY_TIME));
+
+		return;
+	} else if (gps_work.type == GPS_WORK_STOP) {
+		err = stop();
+		if (err) {
+			LOG_ERR("GPS could not be stopped, error: %d", err);
+			return;
+		}
+
+		LOG_INF("GPS operation was stopped");
+
+		atomic_set(&gps_is_active, 0);
+
+		if (atomic_get(&gps_is_enabled) == 0) {
+			return;
+		}
+
+		gps_work.type = GPS_WORK_START;
+
+
+		LOG_INF("The device will try to get fix again in %d seconds\n",
+			CONFIG_GPS_CONTROL_FIX_CHECK_INTERVAL);
+
+		k_delayed_work_submit(&gps_work.work,
+			K_SECONDS(CONFIG_GPS_CONTROL_FIX_CHECK_INTERVAL));
+	}
+}
+
+bool gps_control_is_active(void)
+{
+	return atomic_get(&gps_is_active);
+}
+
+bool gps_control_is_enabled(void)
+{
+	return atomic_get(&gps_is_enabled);
+}
+
+void gps_control_enable(void)
+{
+	atomic_set(&gps_is_enabled, 1);
+	gps_control_start(K_SECONDS(1));
+}
+
+void gps_control_disable(void)
+{
+	atomic_set(&gps_is_enabled, 0);
+	gps_control_stop(K_NO_WAIT);
+}
+
+void gps_control_stop(u32_t delay_ms)
+{
+	k_delayed_work_cancel(&gps_work.work);
+	gps_work.type = GPS_WORK_STOP;
+	k_delayed_work_submit(&gps_work.work, delay_ms);
+}
+
+void gps_control_start(u32_t delay_ms)
+{
+	k_delayed_work_cancel(&gps_work.work);
+	gps_work.type = GPS_WORK_START;
+	k_delayed_work_submit(&gps_work.work, delay_ms);
+}
+
+void gps_control_on_trigger(void)
+{
+
+}
+
+/** @brief Configures and starts the GPS device. */
+int gps_control_init(gps_trigger_handler_t handler)
+{
+	int err;
+	struct device *gps_dev;
+
+	struct gps_trigger gps_trig = {
+		.type = GPS_TRIG_FIX,
+		.chan = GPS_CHAN_NMEA
+	};
+
+	gps_dev = device_get_binding(CONFIG_GPS_DEV_NAME);
+	if (gps_dev == NULL) {
+		LOG_ERR("Could not get %s device", CONFIG_GPS_DEV_NAME);
+		return -ENODEV;
+	}
+
+	err = gps_trigger_set(gps_dev, &gps_trig, handler);
+	if (err) {
+		LOG_ERR("Could not set trigger, error code: %d", err);
+		return err;
+	}
+
+	k_delayed_work_init(&gps_work.work, gps_work_handler);
+
+	gps_work.dev = gps_dev;
+	LOG_DBG("GPS initialized");
+
+	return 0;
+}
+
+/* END: code adapted from applications\asset_tracker\src\gps_controller */
+
+static void send_gps_data_work_fn(struct k_work *work)
+{
+#if 0
```

Comment:
Please remove dead code from the final version.

Reply:
Added CONFIG_THIN_GPS_NOTIFY_NMEA to control result reporting